api/metrics/multi_gatherer.go

```diff
@@ -91,7 +91,14 @@ func (g *multiGatherer) Register(namespace string, gatherer prometheus.Gatherer)
 }
 
 func sortMetrics(m []*dto.MetricFamily) {
-	slices.SortFunc(m, func(i, j *dto.MetricFamily) bool {
-		return *i.Name < *j.Name
+	slices.SortFunc(m, func(i, j *dto.MetricFamily) int {
+		switch {
+		case *i.Name < *j.Name:
+			return -1
+		case *i.Name > *j.Name:
+			return 1
+		default:
+			return 0
+		}
```

Comment:
There are std lib `Compare` and `Less` functions in `package cmp`: https://pkg.go.dev/cmp
```suggestion
		return cmp.Compare(*i.Name, *j.Name)
```

Reply:
Hm, I don't think the `cmp` package is available for `go1.20`

Reply:
I ended up backporting `Compare` for this change with a note to remove it after we update the golang version. Thanks for pointing that out